connectivity/perf/common/metrics.go

```diff
@@ -0,0 +1,168 @@
+			// Let's only export percentiles
+			// Max is skewing results and doesn't make much sense to keep track of
+			"Perc50": float64(metric.Perc50) / float64(time.Microsecond),
+			"Perc90": float64(metric.Perc90) / float64(time.Microsecond),
+			"Perc99": float64(metric.Perc99) / float64(time.Microsecond),
+		},
+		Unit:   "us",
+		Labels: labels,
+	}
+}
+
+type TransactionRateMetric struct {
+	TransactionRate float64 `json:"Rate"` // Ops per second
+}
+
+func (metric *TransactionRateMetric) ToPerfData(labels map[string]string) DataItem {
+	return DataItem{
+		Data: map[string]float64{
+			"Throughput": metric.TransactionRate,
+		},
+		Unit:   "ops/s",
+		Labels: labels,
+	}
+}
+
+type ThroughputMetric struct {
+	Throughput float64 `json:"Throughput"` // Throughput in bytes/s
+}
+
+func (metric *ThroughputMetric) ToPerfData(labels map[string]string) DataItem {
+	return DataItem{
+		Data: map[string]float64{
+			"Throughput": metric.Throughput / 1000000,
+		},
+		Unit:   "Mb/s",
+		Labels: labels,
+	}
+}
+
+type PerfResult struct {
+	Timestamp             time.Time
+	Latency               *LatencyMetric
+	TransactionRateMetric *TransactionRateMetric
+	ThroughputMetric      *ThroughputMetric
+}
+
+type PerfTests struct {
+	Tool     string
+	Test     string
+	SameNode bool
+	Scenario string
+	Sample   int
+	Duration time.Duration
+}
+
+type PerfSummary struct {
+	PerfTest PerfTests
+	Result   PerfResult
+}
+
+// These two structures are borrowed from kubernetes/perf-tests:
+// https://github.com/kubernetes/perf-tests/blob/master/clusterloader2/pkg/measurement/util/perftype.go
+// this is done in order to be compatible with perfdash
+type DataItem struct {
+	// Data is a map from bucket to real data point (e.g. "Perc90" -> 23.5). Notice
+	// that all data items with the same label combination should have the same buckets.
+	Data map[string]float64 `json:"data"`
+	// Unit is the data unit. Notice that all data items with the same label combination
+	// should have the same unit.
+	Unit string `json:"unit"`
+	// Labels is the labels of the data item.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+// PerfData contains all data items generated in current test.
+type PerfData struct {
+	// Version is the version of the metrics. The metrics consumer could use the version
+	// to detect metrics version change and decide what version to support.
+	Version   string     `json:"version"`
+	DataItems []DataItem `json:"dataItems"`
+	// Labels is the labels of the dataset.
+	Labels map[string]string `json:"labels,omitempty"`
+}
+
+func getLabelsForTest(summary PerfSummary, metric string) map[string]string {
+	node := "other-node"
+	if summary.PerfTest.SameNode {
+		node = "same-node"
+	}
+	return map[string]string{
+		"scenario":  summary.PerfTest.Scenario,
+		"node":      node,
+		"test_type": summary.PerfTest.Tool + "-" + summary.PerfTest.Test,
+		"metric":    metric,
+	}
+}
+
+func ExportPerfSummaries(summaries []PerfSummary, reporitDir string) {
```

Comment:
I think there is a typo in the `reporitDir` argument, needs to be `reportDir`

Reply:
Fixed.